compiler-cli/src/build_lock.rs

```diff
@@ -36,7 +36,7 @@ impl BuildLock {
         crate::fs::mkdir(&self.directory).expect("Could not create lock directory");
 
         let lock_path = self.directory.join("gleam.lock");
-        let mut file = fslock::LockFile::open(&lock_path).expect("LockFile creation");
+        let mut file = fslock::LockFile::open(&lock_path.to_string()).expect("LockFile creation");
```

Comment:
We should be able to use this as a reference to an OsStr rather than allocate a string and then discard it immediately. Perhaps `lock_path.as_ref()`?